frontend/src/main/java/com/google/tripmeout/frontend/servlets/ServletUtil.java

```diff
@@ -17,37 +16,34 @@
  * Static utilities that provide functionality that may be shared between servlets.
  */
 public final class ServletUtil {
-  private static final FluentLogger logger = FluentLogger.forEnclosingClass();
-
-  public static <T> Optional<T> extractFromRequestBody(
-      HttpServletRequest request, Gson gson, Class<T> clazz) throws IOException {
-    try (Reader reader = request.getReader()) {
-      return Optional.ofNullable(gson.fromJson(reader, clazz));
-    } catch (JsonParseException e) {
-      logger.atWarning().withCause(e).log(
-          "received exception while attempting to parse json from reader!");
-      return Optional.empty();
+  public static <T> T extractFromRequestBody(Reader reader, Gson gson, Class<T> clazz)
+      throws IOException, JsonParseException, EmptyRequestBodyException {
+    T object = gson.fromJson(reader, clazz);
+    if (object != null) {
```

Comment:
Nit (stylistic choice): if you check for `if (object == null)`, you can throw in the `if` block and just `return object` (outside of the `if`). This reduces block nesting and is a common pattern.